File: android/native/emarsys/src/main/java/com/emarsys/maui/DotnetEmarsysConfig.java
package com.emarsys.maui;

import android.app.Application;
import androidx.annotation.NonNull;
import com.emarsys.Emarsys;
import java.util.List;

public class DotnetEmarsysConfig {

    public static @NonNull EMSConfig build(@NonNull Application application, String applicationCode, String merchantId,
                                           List<String> sharedPackageNames, String sharedSecret, boolean enableConsoleLogging) {
        return new EMSConfig(application, applicationCode, merchantId, sharedPackageNames, sharedSecret, enableConsoleLogging);
    }

    public static void changeApplicationCode(String applicationCode, @NonNull CompletionListener completionListener) {
        Emarsys.getConfig().changeApplicationCode(applicationCode, completionListener::onCompleted);
    }

    public static void changeMerchantId(String merchantId, @NonNull CompletionListener completionListener) {
        Emarsys.getConfig().changeMerchantId(merchantId);
        completionListener.onCompleted(null);
    }

    public static String getApplicationCode() {
        return Emarsys.getConfig().getApplicationCode();
    }

    public static String getMerchantId() {
        return Emarsys.getConfig().getMerchantId();
    }

    public static @NonNull String getClientId() {
        return Emarsys.getConfig().getHardwareId();
    }

    public static @NonNull String getLanguageCode() {
        return Emarsys.getConfig().getLanguageCode();
    }

    public static @NonNull String getSdkVersion() {
        return Emarsys.getConfig().getSdkVersion();
    }

    public static Integer getContactFieldId() {
        return Emarsys.getConfig().getContactFieldId();
    }

}

File: ios/native/MauiEmarsys/DotnetEmarsysConfig.swift
//
//  Copyright © 2024 Emarsys. All rights reserved.
//

import Foundation
import EmarsysSDK

@objc(DotnetEmarsysConfig)
public class DotnetEmarsysConfig: NSObject {
    
    @objc
    public static func build(_ applicationCode: String?, _ merchantId: String?,
                             _ sharedKeychainAccessGroup: String?, _ enableConsoleLogging: Bool = false) -> EMSConfig {
        let config = EMSConfig.make { (build) in
            if let applicationCode = applicationCode, applicationCode != "" {
                build.setMobileEngageApplicationCode(applicationCode)
            }
            if let merchantId = merchantId, merchantId != "" {
                build.setMerchantId(merchantId)
            }
            if let sharedKeychainAccessGroup = sharedKeychainAccessGroup, sharedKeychainAccessGroup != "" {
                build.setSharedKeychainAccessGroup(sharedKeychainAccessGroup)
            }
            if enableConsoleLogging {
                build.enableConsoleLogLevels([EMSLogLevel.basic, EMSLogLevel.error, EMSLogLevel.info, EMSLogLevel.debug])
            }
        }
        return config
    }
    
    @objc
    public static func changeApplicationCode(_ applicationCode: String?, _ completionBlock: @escaping CompletionBlock) {
        Emarsys.config.changeApplicationCode(applicationCode: applicationCode, completionBlock: completionBlock)
    }
    
    @objc
    public static func changeMerchantId(_ merchantId: String?, _ completionBlock: @escaping CompletionBlock) {
        Emarsys.config.changeMerchantId(merchantId: merchantId, completionBlock: completionBlock)
    }
    
    @objc
    public static func getApplicationCode() -> String? {
        return Emarsys.config.applicationCode()
    }
    
    @objc
    public static func getMerchantId() -> String? {
        return Emarsys.config.merchantId()
    }
    
    @objc
    public static func getClientId() -> String {
        return Emarsys.config.hardwareId()
    }
    
    @objc
    public static func getLanguageCode() -> String {
        return Emarsys.config.languageCode()
    }
    
    @objc
    public static func getSdkVersion() -> String {
        return Emarsys.config.sdkVersion()
    }
    
    @objc
    public static func getContactFieldId() -> NSNumber? {
        return Emarsys.config.contactFieldId()
    }
    
}
